src/ipc/ipc4/dai.c

```diff
@@ -36,7 +36,7 @@ void dai_set_link_hda_config(uint16_t *link_config,
 			     struct ipc_config_dai *common_config,
 			     const void *spec_config)
 {
-#if defined(CONFIG_ACE_VERSION_2_0)
+#if defined(CONFIG_ACE_VERSION_2_0) || defined(CONFIG_ACE_VERSION_3_0)
```

Comment:
I don;'t like constructions like this. The list will keep growing with each platform like in "the snake" game. 
Can't it be replaced by some #if CONFIG_FEATURE_NAME ?

Reply:
For CAVS we had CAVS_VERSION kconfig and in such cases it could be used for 
#if CAVS_VERSION >= CONFIG_CAVS_VERSION_2_0

maybe we should introduce similar kconfigs for ACE_VERSION, probably in separate PR

Reply:
I've looked into code of tgl or mtl and there is a gap between https://github.com/thesofproject/sof/blob/main/src/platform/intel/cavs/include/cavs/version.h and https://github.com/thesofproject/sof/blob/main/src/platform/intel/ace/include/ace/version.h . Imho it is better to leave it as is and add some logic to version.h header file.